tests/plugins/test_plugins_manager.py

```diff
@@ -121,6 +121,39 @@ class TestNonPropertyHook(BaseHook):
             self.assertIn('PluginPropertyOperator', str(plugins_manager.operators_modules[0].__dict__))
             self.assertIn("TestNonPropertyHook", str(plugins_manager.hooks_modules[0].__dict__))
 
+    def test_should_fail_on_register_plugin_with_incorrect_load(self):
+        class TestNonPropertyHook(BaseHook):
+            pass
+
+        class AirflowTestPlugin(AirflowPlugin):
+            name = "test_plugin"
+
+            hooks = []
+
+            @classmethod
+            def on_load(cls, *args, **kwargs):
+                cls.hooks.append(TestNonPropertyHook)
+
+        with mock_plugin_manager(plugins=[]):
+            from airflow import plugins_manager
+
+            plugins_manager.register_plugin(AirflowTestPlugin)
+            self.assertListEqual(plugins_manager.plugins, [AirflowTestPlugin])
+
+    def test_should_register_plugin(self):
```

Comment:
```suggestion
    def test_should_fail_on_register_plugin_with_incorrect_load(self):
```

Reply:
Thanks, mixed up the names of the methods